### PR TITLE
Add prefixes to subnet

### PR DESCRIPTION
**Pull request by Am018**
## Description

Added support for multiple address prefixes in subnet

## Checklist

- [Y] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [Y ] The title of the PR is clear and informative
- [Y] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [Y] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [Y] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [Y] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [Y ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: ModuleMappings.json [hunk 1]
```diff
@@ -0,0 +1,269 @@
+{
```
Comment (praries880) [inline ModuleMappings.json]: delete this file

File: SolutionMappings.json [hunk 2]
```diff
@@ -0,0 +1,374 @@
+{
```
Comment (praries880) [inline SolutionMappings.json]: delete this file

File: TestMappings.json [hunk 3]
```diff
@@ -0,0 +1,434 @@
+{
```
Comment (praries880) [inline TestMappings.json]: delete this file

File: tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv [hunk 4]
```diff
@@ -29,3 +29,24 @@
...
 "Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.Automation.GetAzureRMNetworkWatcherReachabilityReport","Get-AzureRMNetworkWatcherReachabilityReport","0","1000","The cmdlet 'Get-AzureRMNetworkWatcherReachabilityReport' has been removed and no alias was found for the original cmdlet name.","Add the cmdlet 'Get-AzureRMNetworkWatcherReachabilityReport' back to the module, or add an alias to the original cmdlet name."
+"Microsoft.Azure.Commands.Network.dll","Microsoft.Azure.Commands.Network.AddAzureApplicationGatewayAuthenticationCertificateCommand","Add-AzureRmApplicationGatewayAuthenticationCertificate","0","3000","The type of property 'AddressPrefix' of type 'Microsoft.Azure.Commands.Network.Models.PSSubnet' has changed from 'System.String' to 'System.Collections.Generic.List`1[System.String]'.","Change the type of property 'AddressPrefix' back to 'System.String'."
```
Comment (praries880) [inline tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv]: Are all of these cmdlets affected by this change?

**Comment by praries880** [comment]
@chandrasekarsrinivasan  can you look at the dll that is being added with this change for network SDK, I am not sure how it will affect the other PRs. Do you guys instead want to check in one that has all the API changes the various teams need?

**Comment by praries880** [comment]
@Am018 Tests are failing because of failure to load the network SDK dll.

example : 
https://travis-ci.org/Azure/azure-powershell/builds/425945125#L1587

Did you generate a signed version of the dll?

**Comment by praries880** [comment]
@Am018 Why are there breaking changes on cmdlets that are not called out in the changelog by this PR? Do we need to update the help files for all those cmdlets as well?

File: tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv (hunk 4 above)
Comment (Am018) [reply tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv]: That's correct. Because all those cmdlets refer to PSSubnet class.

**Comment by praries880** [comment]
on demand : https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/1089

**Comment by cormacpayne** [comment]
@Am018 you will need to make sure that the version of the Network SDK referenced in other projects is bumped as well (Compute, Dns, RecoveryServices, Sql)

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="3.4.0-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.3.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="20.0.0-preview" />
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj]: .NET SDK is merged and (with a bit of magic) finally published https://www.nuget.org/packages/Microsoft.Azure.Management.Network/19.4.0-preview 

So now all contributors must change SDK version from local version (localFeed folder) to 19.4.0 and remove nuget packages from localFeed folder.

**Comment by Am018** [comment]
@cormacpayne @praries880 I have taken care of all the comments by merging with latest changes, removing the csproj files, removing the custom nupkg file. Can you please let me know what else is needed for the completion of the PR?

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md [hunk 6]
```diff
@@ -22,6 +22,7 @@
 
 ## Version 6.7.0
 * Updated cmdlet New-AzureRmVirtualNetworkGatewayConnection with support for switch ExpressRouteGatewayBypass
+* Updated cmdlet New-AzureRmVirtualNetworkSubnetConfig, Set-AzureRmVirtualNetworkSubnetConfig, Add-AzureRmVirtualNetworkSubnetConfig, Get-AzureRmVirtualNetworkSubnetConfig, Add-AzureRmApplicationGatewayAuthenticationCertificate, Add-AzureRmApplicationGatewayFrontendIPConfig, New-AzureRmApplicationGatewayFrontendIPConfig, Set-AzureRmApplicationGatewayFrontendIPConfig, Add-AzureRmApplicationGatewayIPConfiguration, New-AzureRmApplicationGatewayIPConfiguration, Set-AzureRmApplicationGatewayIPConfiguration, Add-AzureRmNetworkInterfaceIpConfig, New-AzureRmNetworkInterfaceIpConfig, Set-AzureRmNetworkInterfaceIpConfig, New-AzureRmVirtualNetworkGatewayIpConfig, Add-AzureRmVirtualNetworkGatewayIpConfig, Set-AzureRmLoadBalancerFrontendIpConfig, Add-AzureRmLoadBalancerFrontendIpConfig, New-AzureRmLoadBalancerFrontendIpConfig and New-AzureRmNetworkInterface by adding support for multiple AddressPrefixes in subnet
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network/ChangeLog.md]: move this under `Current Release`

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkTests.ps1 [hunk 7]
```diff
@@ -187,6 +187,95 @@ function Test-subnetCRUD
     }
 }
 
+<#
+.SYNOPSIS
+Tests creating new simple virtualNetwork and subnets.
+.DESCRIPTION
+SmokeTest
+#>
+function Test-multiPrefixSubnetCRUD
```
Comment (praries880) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkTests.ps1]: you'll need to add the test recording to the desktop test csproj file, [example](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.Netcore.csproj).

File: src/ResourceManager/Network/Commands.Network/ChangeLog.md (hunk 6 above)
Comment (Am018) [reply src/ResourceManager/Network/Commands.Network/ChangeLog.md]: Done!

**Pull request merged**